src/rules/maxLineLengthRule.ts

```diff
@@ -44,20 +71,71 @@ export class Rule extends Lint.Rules.AbstractRule {
         return `Exceeds maximum line length of ${lineLimit}`;
     }
 
-    public isEnabled(): boolean {
-        return super.isEnabled() && this.ruleArguments[0] as number > 0;
-    }
-
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        return this.applyWithFunction(sourceFile, walk, parseOptions(this.ruleArguments));
     }
 }
 
-function walk(ctx: Lint.WalkContext<number>) {
-    const limit = ctx.options;
+function walk(ctx: Lint.WalkContext<Options>) {
     for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength > limit) {
-            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(limit));
+        if (
+            line.contentLength > ctx.options.limit &&
+            !(
+                ctx.options.ignoreLongUrls &&
+                hasUrl(line, ctx.sourceFile) &&
+                lineLengthMinusUrl(
+                    line,
+                    ctx.sourceFile,
+                    Math.round(ctx.options.limit / 2),
+                ) <= ctx.options.limit
+            )
+        ) {
+            ctx.addFailureAt(
+                line.pos,
+                line.contentLength,
+                Rule.FAILURE_STRING_FACTORY(ctx.options.limit),
+            );
+        }
+    }
+}
+
+function buildOptions(options: Partial<Options>): Options {
+    options.ignoreLongUrls =
+        options.ignoreLongUrls === undefined ? false : options.ignoreLongUrls;
+    options.limit = options.limit === undefined ? DEFAULT_LIMIT : options.limit;
```

Comment:
consider creating a new object instead of modifying the original. maybe the caller wants to reuse the object later (unlikely but possible)